Realm/RealmSwift/Object.swift

```diff
@@ -19,40 +19,268 @@
+    // MARK: Properties
+
+    // FIXME: Rename to `realm`
+    /// The `Realm` this object belongs to, or `nil` if the object
+    /// does not belong to a realm (the object is standalone).
+    public var swiftRealm: Realm? {
+        if let rlmRealm = realm {
+            return Realm(rlmRealm)
+        }
+        return nil
     }
 
-    public override init(objectSchema: RLMObjectSchema) {
-        super.init(objectSchema: objectSchema)
+    // FIXME: Rename to `objectSchema`
+    /// The `ObjectSchema` which lists the persisted properties for this object.
+    public var swiftObjectSchema: ObjectSchema { return ObjectSchema(objectSchema) }
+
+    // FIXME: Rename to `invalidated`
+    /// Indicates if an object can no longer be accessed.
+    public var swiftInvalidated: Bool { return invalidated }
+
+    // MARK: Initializers
+
+    /**
+    Initialize a standalone (unpersisted) Object.
+    Call `add(_:)` on a `Realm` to add standalone objects to a realm.
+
+    :see: Realm().add(_:)
+    */
+    public override init() {
+        super.init()
     }
 
-    // And overriding that hides these
+    /**
+    Initialize a standalone (unpersisted) `Object` with values from an `Array<AnyObject>` or `Dictionary<String, AnyObject>`.
+    Call `add(_:)` on a `Realm` to add standalone objects to a realm.
+
+    :param: object The object used to populate the object. This can be any key/value coding compliant
+                   object, or a JSON object such as those returned from the methods in `NSJSONSerialization`,
+                   or an `Array` with one object for each persisted property. An exception will be
+                   thrown if any required properties are not present and no default is set.
+    */
     public override init(object: AnyObject) {
         super.init(object: object)
     }
 
-    public override init() {
-        super.init()
+    // MARK: Constructors
+
+    /**
+    Create an `Object` in the given `Realm` with the given object.
+    
+    Creates an instance of this object and adds it to the given `Realm` populating
+    the object with the given object.
+
+    :param: object The object used to populate the object. This can be any key/value coding compliant
+                   object, or a JSON object such as those returned from the methods in `NSJSONSerialization`,
+                   or an `Array` with one object for each persisted property. An exception will be
+                   thrown if any required properties are not present and no default is set.
+
+                   When passing in an `Array`, all properties must be present,
+                   valid and in the same order as the properties defined in the model.
+    :param: realm  The Realm in which this object is persisted.
+                   The default Realm will be used if this argument is omitted.
+
+    :returns: The created object.
+    */
+    public class func createWithObject(object: AnyObject, inRealm realm: Realm = defaultRealm()) -> Self {
```

Comment:
I agree it's a bit more awkward:

```swift
LargeModel.createWithObject(
  [
    "someLongPropertyA": [
      "aNestedProperty": 123567,
      "anotherNestedProperty": "asdfkjhasdfk jhasdfjkh asdkfjhas fkhasdf"
    ],
    "prop2": "another value"
  ],
  inRealm: realm)
```

```swift
LargeModel.createInRealm(realm, withObject:
  [
    "someLongPropertyA": [
      "aNestedProperty": 123567,
      "anotherNestedProperty": "asdfkjhasdfk jhasdfjkh asdkfjhas fkhasdf"
    ],
    "prop2": "another value"
  ])
```

The best alternative IMO would be to split in two class methods: `RLMObject.createInRealm(_:withObject:)` and `RLMObject.createWithObject(_:)`. However, I still prefer my suggestion because it's a single method, which helps with autocomplete and keeps our API more concise, it's self-documenting and there's a bit less magic IMO.